includes/class-wc-stripe-intent-controller.php

```diff
@@ -733,12 +733,18 @@ public function create_and_confirm_payment_intent( $payment_information ) {
 			$request['setup_future_usage'] = 'off_session';
 		}
 
-		$payment_intent = WC_Stripe_API::request_with_level3_data(
-			$request,
-			'payment_intents',
-			$payment_information['level3'],
-			$order
-		);
+		// Check if a pending payment intent for the same group of params exists to update it instead of creating a new one.
+		$payment_intent = $this->query_for_compatible_payment_intent( $payment_information['customer'], $order->get_id(), $payment_method_types );
```

Comment:
How about checking whether we want to reuse a payment intent _before_ calling this method, and only call it if we want to _create_ one?

I'd try to keep the methods in this class focused on doing a single thing. Currently, this method is in charge of creating a payment intent _and_ retrieving an existing one. If we could make the methods handle a single action, they stay small and focused, they can be reused in different scenarios, and unit-testing them gets simpler. It might not be convenient for every single scenario, but I think it'd be for this use case 🙂 

For example, we could have a different method for retrieving the payment intent object from Stripe, and another method for updating an existing one. The logic for deciding which one to call would live outside of them.

Reply:
With the latest changes, which made the flow a bit simpler, do you think splitting the method in 2 is still necessary? Asking because, right now, the request body for creating and updating is the same.

Reply:
Ok, I found a way to make it still look nice. I created a new method to build the parameters for both create and update methods.